spec/models/activity_spec.rb

```diff
@@ -140,16 +140,20 @@
       let(:product) { activity.price_list.products.sample }
       let(:product_price) { activity.price_list.product_price_for(product).price }
       let(:cash_order) { FactoryBot.create(:order, :cash, activity: activity) }
+      let(:pin_order) { FactoryBot.create(:order, :pin, activity: activity) }
       let(:order) { FactoryBot.create(:order, activity: activity) }
 
       before do
         FactoryBot.create(:order_row, product: product, order: cash_order, product_count: 2)
+        FactoryBot.create(:order_row, product: product, order: pin_order, product_count: 2)
         FactoryBot.create(:order_row, product: product, order: order, product_count: 3)
       end
 
       it { expect(activity.revenue_with_cash).to eq product_price * 2 }
-      it { expect(activity.revenue_without_cash).to eq product_price * 3 }
-      it { expect(activity.revenue_with_cash).not_to eq activity.revenue_without_cash }
+      it { expect(activity.revenue_with_pin).to eq product_price * 2 * 0.981 }
+      it { expect(activity.pin_transaction_fee).to eq product_price * 2 * 0.019 }
+      it { expect(activity.revenue_with_credit).to eq product_price * 3 }
+      it { expect(activity.revenue_with_cash).not_to eq activity.revenue_with_credit }
```

Comment:
Ofcourse this won't eq right? @cmitz what did you try to test here?

Reply:
I don't know, it was more than a year ago :P